### ISSUE TITLE
Resource Exhausted Error.

### ISSUE THREAD

**Issue by tfdeepseeker** [tensorflow/tensorflow#12352]
### This is my first experience with git hub, so, please mention if i didn't provide the right details
Here is the Screen shot of the error.
I am running my program on Tesla K20c GPU which has the following properties:
**name: Tesla K20c
major: 3 minor: 5 memoryClockRate (GHz) 0.7055
pciBusID 0000:41:00.0
Total memory: 4.63GiB
Free memory: 4.57GiB**

![res](https://user-images.githubusercontent.com/31095828/29406999-6f69a618-8360-11e7-97c3-1b36b40503f3.png)

------------------------

### System information
- **OS Platform and Distribution ( Linux Ubuntu 16.04)**:
- **TensorFlow installed from (binary)**:
- **TensorFlow version (0.10.0)**:
- **Python version(3.5.2)**: 

### Describe the problem
I am trying to train the model from scratch and have about 500 train and test images and i am using the batch size as 128. the size of the each image is 512*512.

I tried changing batch size to even 1, but still it is not working.
Why is it showing that error?

Thanks.

**Comment by carlthome** [comment]
Your GPU must have enough memory for not only your data, but also the model (weights and gradients in particular, but also some extra space for computing the operations). Either make your images smaller or reduce the number of layers. You might find [the profiler](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/core/profiler/README.md) helpful.

Also, your question is better suited for [StackOverflow](https://stackoverflow.com/tags/tensorflow).

**Comment by reedwm** [comment]
This question is better asked on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a  bug or feature request. There is also a larger community that reads questions there. Thanks!
